### ISSUE TITLE
Validate tool input

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#233035]
We should validate the input/parameters of LanguageModelTool calls. We could extract part of the json language-service into vscode to do this.

**Comment by aeschli** [comment]
Is this something the tool can do?

**Comment by roblourens** [comment]
Yes, but since the tool already declared its JSON schema to us, it would be a really nice service for vscode to provide

**Comment by roblourens** [comment]
I don't think they should validate against things that the schema can't express- otherwise how does the LLM know about it? 

This will be less important once we have strict function calling though. Hoping that comes soon 🤞

**Comment by roblourens** [comment]
Actually I don't think strict function calling with save us because I don't think other models have this. I think I will end up implementing a retry loop when invalid tool calls are generated.

**Comment by aeschli** [comment]
There's now a command that lets you validate JSON: See https://github.com/microsoft/vscode/pull/244784

**Comment by roblourens** [comment]
Awesome, thanks. Reopening for me to adopt
